tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py

```diff
@@ -0,0 +1,241 @@
+
+
+# Create test vectors for following 9 Params
+# Kernels [2, 3, 5, 7]
+# Input Channels [[3, 4, 16], [17, 25, 33], [40, 48, 64], [80, 96, 100], [160, 512, 640]]
+# Output Channels [3, 4, 16, 40, 64, 96, 100, 150, 512, 1024]
+# Transpose_mcast
+# output_layout
+# has_bias
+# enable_act_double_buffer
+# enable_split_reader
+# enable_subblock_padding
+
+# Keeping rest of the params constant as they do not affect sharding
+# Batches
+# Activations
+# Dilation
+# Activation Datatype
+# Weight Datatypes
+# Groups
+
+# Total test cases = 4 * 15 * 10 * 2 * 2 * 2 * 2 * 2 * 2 = 38400
+# Total 5(7680 each) suites are created since currently, only 10K test vectors can be generated per suite.
+# Note that some test cases might be invalid.
+
+input_channels = [[3, 4, 16], [17, 25, 33], [40, 48, 64], [80, 96, 100], [160, 512, 640]]
+
+parameters = {
+    f"height_sharded_suite_{idx}": {
+        "height_sharded_specs": list(
+            get_height_sharded_specs(
+                [2],  # batch_sizes
+                [32],  # activation size
+                [2, 3, 5, 7],  # kernel sizes
+                [1],  # stride sizes
+                [1],  # padding sizes
+                [1],  # dilations sizes
+            )
+        ),
+        "input_channels": channels,
+        "output_channels": [3, 4, 16, 40, 64, 96, 100, 150, 512, 1024],
+        "transpose_mcast": [True, False],
+        "output_layout": [ttnn.TILE_LAYOUT, ttnn.ROW_MAJOR_LAYOUT],
+        "has_bias": [True, False],
+        "enable_act_double_buffer": [True, False],
+        "enable_split_reader": [True, False],
+        "enable_subblock_padding": [True, False],
+        "activations_dtype": [ttnn.bfloat16],
+        "weights_dtype": [ttnn.bfloat16],
+        "groups": [1],
+        "math_fidelity": [ttnn.MathFidelity.HiFi4],
+        "config_override": [None],
+        "use_shallow_conv_variant": [False],
+        "enable_auto_formatting": [False],
+        "padded_input_channels": [None],
+        "fp32_accum": [False],
+        "packer_l1_acc": [False],
+        "deallocate_activation": [False],
+        "debug": [False],
+    }
+    for idx, channels in enumerate(input_channels)
+}
+
+
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    return False, None
+
+
+def mesh_device_fixture():
+    num_devices = ttnn.GetNumPCIeDevices()
+    # As of now take device id as 0.
+    device_id = 0
+    assert device_id < num_devices, "CreateDevice not supported for non-mmio device"
+    device = ttnn.CreateDevice(device_id=device_id, l1_small_size=32768)
+    ttnn.SetDefaultDevice(device)
+
+    ARCH_NAME = os.environ.get("ARCH_NAME", os.environ.get("TT_ARCH_NAME", "")).lower()
+    device_name = "Unknown"
+    if ARCH_NAME == "grayskull":
+        device_name = "grayskull"
+    elif ARCH_NAME == "wormhole_b0":
+        device_name = "wormhole_b0"
+    yield device, device_name
+
+    ttnn.synchronize_device(device)
```

Comment:
i think we dont need to explicitly sync before closing the device.

Reply:
Done.